### PR TITLE
Update webpack & co.

### PR DESCRIPTION
**Pull request by T4rk1n**

### CODE REVIEW

File: {{cookiecutter.project_shortname}}/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
 {%- endif %}
   "main": "build/index.js",
   "scripts": {
-    "start": "webpack-serve --config ./webpack.serve.config.js --open",
+    "start": "webpack serve --config ./webpack.serve.config.js --open",
```
Comment (alexcjohnson) [inline {{cookiecutter.project_shortname}}/package.json:17]: Do people still use the demo? I wonder if we'd be better off removing it, so we only explicitly support testing components inside a Dash app.

File: {{cookiecutter.project_shortname}}/package.json [hunk 2]
```diff
@@ -29,30 +29,29 @@
...
     "react-dom": "^16.8.6",
-    "styled-jsx": "^3.2.1",
     "style-loader": "^0.23.1",
-    "terser-webpack-plugin": "^2.3.0",
-    "webpack": "4.36.1",
-    "webpack-cli": "3.3.6",
-    "webpack-serve": "3.1.0"
+    "styled-jsx": "^3.2.1",
+    "webpack": "5.84.1",
+    "webpack-cli": "5.1.1",
```
Comment (alexcjohnson) [inline {{cookiecutter.project_shortname}}/package.json:53]: Is there a reason these two have no `^`?
```suggestion
    "webpack": "^5.84.1",
    "webpack-cli": "^5.1.1",
```

File: {{cookiecutter.project_shortname}}/package.json [hunk 3]
```diff
@@ -29,30 +29,29 @@
...
     "eslint-plugin-import": "^2.18.0",
     "eslint-plugin-react": "^7.14.2",
-    "prop-types": "^15.7.2",
+    "prop-types": "^15.8.1",
     "react": "^16.8.6",
-    "react-docgen": "^4.1.1",
+    "react-docgen": "^5.4.3",
     "react-dom": "^16.8.6",
-    "styled-jsx": "^3.2.1",
     "style-loader": "^0.23.1",
```
Comment (alexcjohnson) [inline {{cookiecutter.project_shortname}}/package.json:50]:
```suggestion
    "style-loader": "^3.3.3",
```
What's the pattern for which packages you updated and which you didn't? Let's try and get a few more of these updated while we're at it. I think if we use the latest `style-loader` and `css-loader` we just need to change the webpack config for css to:
```
use: ['style-loader', 'css-loader']
```

File: {{cookiecutter.project_shortname}}/package.json (hunk 1 above)
Comment (alexcjohnson) [reply {{cookiecutter.project_shortname}}/package.json:17]: OK, we can leave it. Most of our own components the demo is either nonfunctional or has been deleted entirely, but if others use it we can leave it there.

**Comment by alexcjohnson** [comment]
Hmm need to re-enable circleci to clone the repo?

**Comment by T4rk1n** [comment]
> Hmm need to re-enable circleci to clone the repo?

Yes, the config are also using the old image and should be updated to use the `cimg` ones

**Pull request merged**